x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -395,9 +399,20 @@ export const loadFullStoryUserId = async ({
           currentUser.metadata
         )}`
       );
+
+      return undefined;
+    }
+
+    if (
+      getIsCloudEnabled(cloudDeploymentId) &&
+      currentUser.authentication_realm?.name === 'cloud-saml-kibana'
```

Comment:
To be extra safe, I think we'll also want to check that `currentUser.authentication_realm?.type === 'saml'`.

Reply:
Good call! Added!